WordPress/Classes/Utility/ImageURLs/WPImageURLHelper+Photon.swift

```diff
@@ -0,0 +1,172 @@
+//
+
+import Foundation
+
+extension WPImageURLHelper
+{
+    private static let acceptedImageTypes = ["gif", "jpg", "jpeg", "png"]
+
+    private enum PhotonSubdomain: String {
+        case Zero = "i0"
+        case One = "i1"
+        case Two = "i2"
+    }
+
+    private enum PhotonQueryFields: String {
+        case SSL = "ssl"
+        case Quality = "quality"
+        case Resize = "resize"
+        case Fit = "fit"
+        case Crop = "crop"
+        case AddLetterbox = "lb"
+        case RemoveLetterBox = "ulb"
+        case Filter = "filter"
+        case Brightness = "brightness"
+        case Contrast = "contrast"
+        case Colorize = "colorize"
+        case Smooth = "smooth"
+        case Zoom = "zoom"
+        case Strip = "strip"
+    }
+
+    private enum PhotonFilterValue: String {
+        case Negate = "negate"
+        case Grayscale = "grayscale"
+        case Sepia = "sepia"
+        case EdgeDetect = "edgedetect"
+        case Emboss = "emboss"
+        case BlurGaussian = "blurgaussian"
+        case BlurSelective = "blurselective"
+        case MeanRemoval = "meanremoval"
+    }
+
+    private enum PhotonStipValue: String {
+        case All = "all"
+    }
+
+    private enum PhotonSSLValue: String {
+        case Enabled = "1"
+    }
+
+    private enum PhotonImageQuality: UInt {
+        case Max = 100
+        case Default = 80
+        case Min = 1
+    }
+
+    /**
+     Create a "photonized" URL from the passed arguments. Kept as a convenient way to use default values for `forceResize` and `imageQuality` parameters from ObjC.
+
+     - parameters:
+     - size: The desired size of the photon image. If height is set to zero the returned image will have a height proportional to the requested width.
+     - url: The URL to the source image.
+
+     - returns: A URL to the photon service with the source image as its subject.
+     */
+    public static func photonDefaultURL(withSize size: CGSize, forImageURL url: NSURL) -> NSURL? {
+        return photonURL(withSize: size, forImageURL: url, forceResize: true, imageQuality: PhotonImageQuality.Default.rawValue)
+    }
+
+    /**
+     Create a "photonized" URL from the passed arguments.
+
+     - parameters:
+     - size: The desired size of the photon image. If height is set to zero the returned image will have a height proportional to the requested width.
+     - url: The URL to the source image.
+     - forceResize: By default Photon does not upscale beyond a certain percentage. Setting this to `true` forces the returned image to match the specified size. Default is `true`.
+     - quality: An integer value 1 - 100. Passed values are constrained to this range. Default is 80%.
+
+     - returns: A URL to the photon service with the source image as its subject.
+     */
+    public static func photonURL(withSize size: CGSize, forImageURL url: NSURL, forceResize: Bool = true, imageQuality: UInt = PhotonImageQuality.Default.rawValue) -> NSURL? {
```

Comment:
This method also seems a bit to big, maybe try to create a couple of sub methods to be able to review it easily?